src/server/cmd/job-shim/main.go

```diff
@@ -47,6 +47,23 @@ func do(appEnvObj interface{}) error {
 				os.Exit(0)
 			}
 
+			// Make sure that we call FinishJob even if something caused a panic
+			defer func() {
+				if r := recover(); r != nil {
+					if _, err := ppsClient.FinishJob(
```

Comment:
This seems unlikely but could we bulletproof this against calling `FinishJob` below, then getting a panic recovering and calling FinishJob again? That could be a nasty bug since the job would finish while one of the pods is still running.